stable_plugins/classical_ml/data_preparation/pandas_preprocessing/routes.py

```diff
@@ -0,0 +1,361 @@
+        required=False,
+    )
+    @PDPreprocessing_BLP.require_jwt("jwt", optional=True)
+    def post(self, errors, db_id: int, step_id: int):
+        """Return the micro frontend with prerendered inputs."""
+        return self.render(request.form, db_id, step_id, errors)
+
+    def render(self, data: Mapping, db_id: int, step_id: int, errors: dict):
+        db_task: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+        if db_task is None:
+            msg = f"Could not load task data with id {db_id} to read parameters!"
+            TASK_LOGGER.error(msg)
+            raise KeyError(msg)
+
+        schema = SecondInputParametersSchema()
+
+        data_dict = dict(data)
+        fields = schema.fields
+        # define default values
+        default_values = {
+            fields["threshold"].data_key: 0,
+        }
+
+        # overwrite default values with other values if possible
+        default_values.update(data_dict)
+        data_dict = default_values
+
+        return Response(
+            render_template(
+                "second_template.html",
+                name=PDPreprocessing.instance.name,
+                version=PDPreprocessing.instance.version,
+                schema=schema,
+                values=data_dict,
+                errors=errors,
+                process=url_for(
+                    f"{PDPreprocessing_BLP.name}.FinalProcessView",
+                    db_id=db_id,
+                    step_id=step_id,
+                ),
+                add_step=url_for(
+                    f"{PDPreprocessing_BLP.name}.SecondProcessView",
+                    db_id=db_id,
+                    step_id=step_id,
+                ),
+                pandas_html=Markup(db_task.data["pandas_html"]),
+                columns_and_rows_html=Markup(db_task.data["columns_and_rows_html"]),
+                columns_as_optionlist=Markup(db_task.data["columns_as_optionlist"]),
+            )
+        )
+
+
+@PDPreprocessing_BLP.route("/<int:db_id>/<int:step_id>-process1/")
+class FinalProcessView(MethodView):
+    """Start a long running processing task."""
+
+    @PDPreprocessing_BLP.arguments(
+        SecondInputParametersSchema(unknown=EXCLUDE), location="form"
+    )
+    @PDPreprocessing_BLP.response(HTTPStatus.OK, TaskResponseSchema())
+    @PDPreprocessing_BLP.require_jwt("jwt", optional=True)
+    def post(self, arguments, db_id: int, step_id: int):
+        """Start the calculation task."""
+        db_task: Optional[ProcessingTask] = ProcessingTask.get_by_id(id_=db_id)
+        if db_task is None:
+            msg = f"Could not load task data with id {db_id} to read parameters!"
+            TASK_LOGGER.error(msg)
+            raise KeyError(msg)
+
+        db_task.parameters = SecondInputParametersSchema().dumps(arguments)
+
+        db_task.clear_previous_step()
+        db_task.save(commit=True)
+
+        # all tasks need to know about db id to load the db entry
+        task: chain = second_task.s(
```

Comment:
The naming is quite confusing with "second_task" being used in "FinalProcess"...

Reply:
Only the naming is confusing here. Naming something purely by the order it appears in (first_, second_, final_) works, but usually does not convey what the individual tasks do. Something like `setup`, `preprocessing`, `final/finishing`  would be less confusing. (good names are hard)